src/polkadot/fellows.ts

```diff
@@ -10,83 +15,121 @@ export class PolkadotFellows implements TeamApi {
+      const peopleRelayChain = await smoldot.addChain({
+        chainSpec: peopleChainSpec,
+        potentialRelayChains: [smoldotRelayChain],
+      });
+
+      // Initialize the smoldot provider
+      const jsonRpcProvider = getSmProvider(peopleRelayChain);
+      logger.info("Initializing the people client");
+      const peopleClient = createClient(jsonRpcProvider);
+
+      // Get the types for the people client
+      const peopleApi = peopleClient.getTypedApi(people);
+
+      const getGhHandle = async (address: SS58String): Promise<string | undefined> => {
+        logger.debug(`Fetching identity of '${address}'`);
+        const identityOf = await peopleApi.query.Identity.IdentityOf.getValue(address);
+
+        if (identityOf) {
+          const [identity] = identityOf;
+          const github = identity.info.github.value;
+
+          if (!github) {
+            logger.debug(`'${address}' does not have an additional field named 'github'`);
+            return;
+          }
+
+          const handle = github.asText().replace("@", "") as string;
+
+          if (handle) {
+            logger.info(`Found github handle for '${address}': '${handle}'`);
           } else {
-            this.logger.debug("No super identity found. Skipping");
+            logger.debug(`'${address}' does not have a GitHub handle`);
+            return;
           }
-          continue;
+          return handle;
         }
 
-        const [fellowData] = identityQuery.toHuman() as [Record<string, unknown>, unknown];
+        logger.debug(`Identity of '${address}' is null. Checking for super identity`);
 
-        // @ts-ignore
-        const additional = fellowData.info?.additional as [{ Raw: string }, { Raw: string }][] | undefined;
+        const superIdentityAddress = (await peopleApi.query.Identity.SuperOf.getValue(address))?.[0];
 
-        // If it does not have additional data (GitHub handle goes here) we ignore it
-        if (!additional || additional.length < 1) {
-          this.logger.debug("Additional data is null. Skipping");
-          continue;
+        if (superIdentityAddress) {
+          logger.debug(`'${address}' has a super identity: '${superIdentityAddress}'. Fetching that identity`);
+          return await getGhHandle(superIdentityAddress);
+        } else {
+          logger.debug(`No superidentity for ${address} found.`);
+          return undefined;
         }
-
-        for (const additionalData of additional) {
-          const [key, value] = additionalData;
-          // We verify that they have an additional data of the key "github"
-          // If it has a handle defined, we push it into the array
-          if (key?.Raw && key?.Raw === "github" && value?.Raw && value?.Raw.length > 0) {
-            this.logger.debug(`Found handles: '${value.Raw}'`);
-            // We add it to the array and remove the @ if they add it to the handle
-            users.set(value.Raw.replace("@", ""), fellow.rank);
-          }
+      };
```

Comment:
Could we move this function outside please? It got too nested so I didn't get where the function ends at first 😅